docs/query-service.md

```diff
@@ -191,3 +195,304 @@ type Config struct {
     MaxViewTimeout        time.Duration
 }
 ```
+
+## 6. Implementation
+
+The Query Service consists of several key components:
+
+* **Service**: The main entry point that implements the gRPC `QueryServiceServer` interface. It handles incoming requests, manages views, and coordinates query execution.
+* **ViewsBatcher**: The component that manages views and batches queries for efficient database access. It uses lock-free data structures to minimize contention and maximize throughput.
+* **ViewHolder**: Represents a client's view of the database with specific isolation requirements. Each view has a unique ID and parameters that determine its behavior.
+* **Batcher**: Aggregates queries with similar parameters to optimize database access. Multiple views with the same parameters may share a batcher.
+* **NamespaceQueryBatch**: Collects keys to be queried within a specific namespace. Once a batch reaches a certain size or age, it is executed as a single database query.
+
+### Implementation Flow
+
+The Query Service operates with the following three stages.
+
+#### Stage 1. View Management
+
+View management provides clients with consistent access to the database through configurable isolation levels.
+This stage is optional as the client can use `nil` view, which does not offer any consistency between future reads.
```

Comment:
When the view is `nil`, is it not equivalent to the committed read? The term consistency might be confusing here with the isolation levels. We can stick to isolation levels alone.

Reply:
Yes. But repeatable reads for future read calls cannot be guaranteed if the view is not used.

Reply:
Added clarification